types/index.d.ts

```diff
@@ -0,0 +1,335 @@
+     *
+     * Please note that multitouch is off when in static or semi modes.
+     */
+    multitouch?: boolean;
+
+    /**
+     * Defaults to `1`
+     *
+     * If you need to, you can also control the maximum number of instance that could be created.
+     *
+     * Obviously in a multitouch configuration.
+     */
+    maxNumberOfNipples?: number;
+
+    /**
+     * Defaults to `false`
+     *
+     * The library won’t draw anything in the DOM and will only trigger events with data.
+     */
+    dataOnly?: boolean;
+
+    /**
+     * Defaults to `{top: 0, left: 0}`
+     *
+     * An object that will determine the position of a static mode.
+     *
+     * You can pass any of the four top, right, bottom and left.
+     *
+     * They will be applied as any css property.
+     */
+    position?: Position;
+
+    /**
+     * Behavioral mode for the joystick
+     *
+     * ### 'dynamic':
+     * a new joystick is created at each new touch.
+     * the joystick gets destroyed when released.
+     * can be multitouch.
+     *
+     * ### 'semi':
+     * new joystick is created at each new touch farther than options.catchDistance of any previously created joystick.
+     * the joystick is faded-out when released but not destroyed.
+     * when touch is made inside the options.catchDistance a new direction is triggered immediately.
+     * when touch is made outside the options.catchDistance the previous joystick is destroyed and a new one is created.
+     * cannot be multitouch.
+     *
+     * ### 'static':
+     * a joystick is positioned immediately at options.position.
+     * one joystick per zone.
+     * each new touch triggers a new direction.
+     * cannot be multitouch.
+     */
+    mode?: 'dynamic' | 'semi' | 'static';
+
+    /**
+     * Defaults to `true`
+     *
+     * Reset the joystick’s position when it enters the rest state.
+     */
+    restJoystick?: boolean;
+
+    /**
+     * Defaults to `0.5`
+     * The opacity to apply when the joystick is in a rest position.
+     */
+    restOpacity?: number;
+
+    /**
+     * Defaults to `200`
+     *
+     * This is only useful in the semi mode, and determine at which distance we recycle the previous joystick.
+     */
+    catchDistance?: number;
+
+    /**
+     * Defaults to `false`
+     *
+     * Locks joystick’s movement to the x (horizontal) axis
+     */
+    lockX?: boolean;
+
+    /**
+     * Defaults to `false`
+     *
+     * Locks joystick’s movement to the y (vertical) axis
+     */
+    lockY?: boolean;
+}
+
+export interface Position {
+    x: number;
+    y: number;
+}
+
+export interface JoystickOutputData {
+    angle: {
+        degree: number;
+        radian: number;
+    };
+    direction: {
+        angle: string;
+        x: string;
+        y: string;
+    };
+    distance: number;
+    force: number;
+    identifier: number;
+    instance: Joystick;
+    position: Position;
+    pressure: number;
+}
+
+export type JoystickEventTypes =
```

Comment:
Decided to add string unions for these event types as well.

Reply:
Thanks, and thank you for the time reviewing!

The module part was the tricky part for me. I'm not 100% sure if this would normally add itself to the window's global name-space, but it wasn't in my web-packed project. The object returned from the IIFE is a small one that invokes a Manager instance, so that's what I put here.